src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -47,23 +62,102 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
             }
         }
 
+        private static ConcurrentDictionary<string, IConsoleLogFormatter> SetFormatters(IEnumerable<IConsoleLogFormatter> fromSetup = null)
+        {
+            IEnumerable<IConsoleLogFormatter> formatters = fromSetup;
+            if (formatters == null || formatters.ToList().Count == 0)
+            {
+                var defaultMonitor = new FormatterOptionsMonitor<DefaultConsoleLogFormatterOptions>(new DefaultConsoleLogFormatterOptions());
+                var systemdMonitor = new FormatterOptionsMonitor<SystemdConsoleLogFormatterOptions>(new SystemdConsoleLogFormatterOptions());
+                formatters = new List<IConsoleLogFormatter>()
+                {
+                    new DefaultConsoleLogFormatter(defaultMonitor),
+                    new SystemdConsoleLogFormatter(systemdMonitor)
+                };
+            }
+            return new ConcurrentDictionary<string, IConsoleLogFormatter>(formatters.ToDictionary(f => f.Name));
+        }
+
+        // warning:  ReloadLoggerOptions can be called before the ctor completed,... before registering all of the state used in this method need to be initialized
         private void ReloadLoggerOptions(ConsoleLoggerOptions options)
         {
-            foreach (System.Collections.Generic.KeyValuePair<string, ConsoleLogger> logger in _loggers)
+            if (
+                options.FormatterName == null ||
+                !_formatters.TryGetValue(options.FormatterName, out IConsoleLogFormatter logFormatter) ||
+                !_formatters.TryGetValue(options.FormatterName?.ToLower(), out logFormatter)
+                )
+            {
+#pragma warning disable CS0618
+                switch (options.Format)
```

Comment:
>Format [](start = 32, length = 6)

can't we get other formats here (e.g. Json)?

Reply:
This code blob only tries to make the old workflow still work. we only had Default and Systemd before.